ruby-challenges.rb

```diff
@@ -10,22 +10,66 @@
 filter_letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+    def filter_method (arr, letter)
+        arr.keep_if{|arr| arr.include?(letter)}
+    end
+
+    # p filter_method(beverages_array, filter_letter_o)
+    # p filter_method(beverages_array, filter_letter_t)
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
 us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Rhode Island'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Rhode Island', 'Washington'] 
 
+    def order_hash hash
+        hash.values.flatten
+    end
+
+    # p order_hash(us_states)
 
 # --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
 
 # Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
+    class Bike
+        attr_accessor :model, :wheels, :current_speed
+
+        def initialize (model)
+            @model = model
+            @wheels = 2
+            @current_speed = 0
+        end
+
+        def bike_info
+            "The #{model} bike has #{wheels} wheels and is going #{current_speed} mph"
+        end
 
+        def pedal_faster(num)
+            @current_speed += num if num.positive? 
+        end
 
+        def braking(num)
+            @current_speed -= num if (@current_speed - num).negative?
```

Comment:
I am not sure why I keep that there after I add 0 if @current_speed.negative?

Reply:
Well here you want to set `@current_speed -= num` if the difference to that equation is >= 0, so the way you have it set up now, if current_speed is 50, and you used .braking 25, current_speed is still going to be 50, because it won't get set to something else since current_speed - num is not negative.